src/external.rs

```diff
@@ -241,10 +241,19 @@ pub fn register_test_contract(e: &Env, contract_id: &U256) {
     e.register_contract(bin.into(), crate::contract::Token {});
 }
 
-pub fn initialize(e: &mut Env, contract_id: &U256, admin: &Identifier) {
+pub fn initialize(
+    e: &mut Env,
+    contract_id: &U256,
+    admin: &Identifier,
+    decimal: &u32,
```

Comment:
Decimals has to be `u8`, according to the `initialize` function in contract.rs.

Reply:
Contracts don't support u8 as a type. u32/i32 is the smallest integer type.

Reply:
I know, but this isn't a `contractfn`. It's an external function.